cmd/openstack-populator/openstack-populator.go

```diff
@@ -155,61 +179,28 @@ func (cr *CountingReader) Read(p []byte) (int, error) {
 	return n, err
 }
 
-func writeData(reader io.ReadCloser, file *os.File, crName, crNamespace string) error {
-	// 	config, err := rest.InClusterConfig()
-	// 	if err != nil {
-	// 		klog.Fatal(err.Error())
-	// 	}
-
-	// 	client, err := dynamic.NewForConfig(config)
-	// 	if err != nil {
-	// 		klog.Fatal(err.Error())
-	// 	}
-	// 	gvr := schema.GroupVersionResource{Group: groupName, Version: apiVersion, Resource: resource}
-
+func writeData(reader io.ReadCloser, file *os.File, imageID, controllerIp string, progress *prometheus.GaugeVec) error {
 	total := new(int64)
 	countingReader := CountingReader{reader, total}
 
-	// TODO introduce /metrics endpoint to report progress
-	// done := make(chan bool)
-	// go func() {
-	// 	for {
-	// 		select {
-	// 		case <-done:
-	// 			return
-	// 		default:
-	// 			populatorCr, err := client.Resource(gvr).Namespace(crNamespace).Get(context.TODO(), crName, metav1.GetOptions{})
-	// 			if err != nil {
-	// 				klog.Fatal(err.Error())
-	// 			}
-	// 			status := map[string]interface{}{"transferred": fmt.Sprintf("%d", *total)}
-	// 			unstructured.SetNestedField(populatorCr.Object, status, "status")
-
-	// 			_, err = client.Resource(gvr).Namespace(crNamespace).Update(context.TODO(), populatorCr, metav1.UpdateOptions{})
-	// 			if err != nil {
-	// 				klog.Fatal(err.Error())
-	// 			}
-	// 		}
-
-	// 		time.Sleep(3 * time.Second)
-	// 	}
-	// }()
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				progress.WithLabelValues(imageID).Set(float64(*total))
+			}
+		}
+	}()
 
 	if _, err := io.Copy(file, &countingReader); err != nil {
 		klog.Fatal(err)
 	}
-	//done <- true
-	// populatorCr, err := client.Resource(gvr).Namespace(crNamespace).Get(context.TODO(), crName, metav1.GetOptions{})
-	// if err != nil {
-	// 	klog.Fatal(err.Error())
-	// }
-	// status := map[string]interface{}{"transferred": *countingReader.total}
-	// unstructured.SetNestedField(populatorCr.Object, status, "status")
-
-	// _, err = client.Resource(gvr).Namespace(crNamespace).Update(context.TODO(), populatorCr, metav1.UpdateOptions{})
-	// if err != nil {
-	// 	klog.Error(err)
-	// }
+	done <- true
+	progress.WithLabelValues(imageID).Set(float64(*total))
```

Comment:
I can actually set to 100% percent here and it would be simpler, but it would be slightly harder for ovirt because we'll need to read the full size from disk or something similar

Reply:
yeah, the actual/apparent size.. I think that for completeness we'll want to keep that "logic" in the controller that sets the transfer to 100% when the populator pod is completed successfully so it's not that important

Reply:
yeah, the actual/apparent size.. I think that for completeness we'll want to keep that "logic" in the controller that sets the transfer to 100% when the populator pod is completed successfully so it's not that important